pkg/model/components/docker.go

```diff
@@ -53,7 +53,11 @@ func (b *DockerOptionsBuilder) BuildOptions(o interface{}) error {
 		}
 
 		dockerVersion := ""
-		if sv.Major == 1 && sv.Minor >= 12 {
+		if sv.Major == 1 && sv.Minor >= 17 {
+			dockerVersion = "19.03.4"
+		} else if sv.Major == 1 && sv.Minor >= 14 {
```

Comment:
While this might be a good idea, we do have a policy of not changing the defaults on existing clusters, and we have shipped kops 1.14.  Would you mind making this >= 15, therefore?

(And we'll have to cherry-pick back to 1.15...)

Otherwise I cross checked and the versions look good, k8s 1.15 supports 18.09, k8s 1.17 adds support for 19.03 - thank you!

Reply:
I still haven't figured out the support policy for Docker, but I have nothing against bumping to >= 15.

From security point of view, it is hard to tell if there are any issues with 18.06.3. Not being an EE release, it is unmaintained since Feb. I didn't see anything that caught my eye in the Docker release notes, but Contained seems to have many "notable updates" including CVE fixes since then.

Reply:
One idea would be to cherry-pick only the package definitions to released versions. This way, any user can upgrade Docker at its own risk.